janusgraph-core/src/main/java/org/janusgraph/graphdb/configuration/GraphDatabaseConfiguration.java

```diff
@@ -1286,15 +1301,54 @@ public GraphDatabaseConfiguration(ReadConfiguration localConfig) {
                 } else {
                     log.info("Using configured timestamp provider {}", localBasicConfiguration.get(TIMESTAMP_PROVIDER));
                 }
-
                 globalWrite.freezeConfiguration();
             } else {
+                final boolean allowUpgrade;
+                String graphName = getGraphName();
+                if (graphName == null) {
+                    graphName = "";
+                }
+
+                if (localBasicConfiguration.has(ALLOW_UPGRADE))
+                    allowUpgrade = localBasicConfiguration.get(ALLOW_UPGRADE);
+                else if (globalWrite.has(ALLOW_UPGRADE))
+                    allowUpgrade = globalWrite.get(ALLOW_UPGRADE);
+                else
+                    allowUpgrade = ALLOW_UPGRADE.getDefaultValue();
+
+                if (allowUpgrade) {
```

Comment:
Can this block be tested to improve coverage (see [here](https://codecov.io/gh/JanusGraph/janusgraph/src/storage-version/janusgraph-core/src/main/java/org/janusgraph/graphdb/configuration/GraphDatabaseConfiguration.java#L1319...1341))?

Reply:
I've been working on improving the test coverage in a different branch. GraphDatabaseConfiguration needs to be refactored to make it possible. Would it be ok if I opened a separate issue to track the refactor and test coverage improvement?

Reply:
Sure thanks for taking a look. There's probably already a separate issue for improving test coverage.